ring/ownership_priority_queue.go

```diff
@@ -0,0 +1,200 @@
+package ring
+
+import (
+	"container/heap"
+	"fmt"
+	"math"
+)
+
+type ringItem interface {
+	GetID() int
+	Less(other ringItem) bool
+}
+
+type ringInstance struct {
+	instanceID int
+}
+
+func newRingInstance(instanceID int) *ringInstance {
+	return &ringInstance{
+		instanceID: instanceID,
+	}
+}
+
+func (ri *ringInstance) GetID() int {
+	return ri.instanceID
+}
+
+func (ri *ringInstance) Less(other ringItem) bool {
+	ringInstance, ok := other.(*ringInstance)
+	if !ok {
+		return false
+	}
+	return ri.instanceID < ringInstance.instanceID
+}
+
+func (ri *ringInstance) String() string {
+	return fmt.Sprintf("[instanceID: %d]", ri.instanceID)
+}
+
+type ringToken struct {
+	token     uint32
+	prevToken uint32
+}
+
+func newRingToken(token, prevToken uint32) *ringToken {
+	return &ringToken{
+		token:     token,
+		prevToken: prevToken,
+	}
+}
```

Comment:
These constructors don't add any value to the code, and they're not idiomatic in golang. 
These are private types and we build them with all fields and we don't do any kind of logic (not even a single `make()` call that could justify them). I'd remove them.